eis_toolkit/vector_processing/proximity_tool.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import geopandas as gpd
+from numbers import Number
+from beartype import beartype
+from beartype.typing import Union
+from rasterio import profiles, transform
+from eis_toolkit.vector_processing.distance_computation import distance_computation
+
+@beartype
+def calculate_proximity(geodataframe: gpd.GeoDataFrame, raster_profile: Union[profiles.Profile, dict], maximum_distance: Number) -> np.ndarray:
+
+    """ Interpolates the distance values calculated by the distance_computation function between 0 and 1.
+        1 denots the value inside the polygon and 0 at the maximum distance.
+        If maximum_distance value is not provided, the program sets this value to the maximum value 
+        in the provided distance matrix.
+        Uses linear interpolation to calculate the distance from the polygon.
+
+        Args:
+            geodataframe: The GeoDataFrame with geometries to determine distance to.
+            raster_profile: The raster profile of the raster in which the distances
+                            to the nearest geometry are determined.
+            max_distance: The maximum distance in the output array.
+
+        Returns:
+            A 2D numpy array with the the distance values inverted.
+    """
+
+    out_matrix = distance_computation(geodataframe, raster_profile, maximum_distance)
+
+    minimum = np.min(out_matrix)
+    difference = maximum_distance - minimum
+    out_matrix = maximum_distance - out_matrix
+    out_matrix = out_matrix/difference
+
+    return out_matrix
+
+@beartype
+def calculate_logarithmic_proximity(geodataframe: gpd.GeoDataFrame, raster_profile: Union[profiles.Profile, dict], maximum_distance: Number) -> np.ndarray:
```

Comment:
I think it would be better if the choice of scaling should be parameterized, not formulated as another public function. So, let's expose only one `proximity_computation` to the users with parameter `scaling`. It could be literal `Literal['linear', 'log']` with linear as default.